### PR TITLE
Expose /api/authenticate endpoint in oidc/skip user management flows

### PR DESCRIPTION
**Pull request by vishal423**
This API is useful to determine on the front end about authentication status of the user. Based on the API response, frontend logic can decide to retrieve the user information or to redirect to the login page, or do any other action. It's already been available under session / jwt authentication types.
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/java/package/web/rest/AccountResource.java.ejs [hunk 1]
```diff
@@ -80,6 +81,24 @@ public class AccountResource {
...
+
+    /**
+     * {@code GET  /authenticate} : check if the user is authenticated, and return its login.
+     *
+     * @param request the HTTP request.
+     * @return the login if the user is authenticated.
+     */
+    @GetMapping("/authenticate")
+    <%_ if (reactive) { _%>
+    public Mono<String> isAuthenticated(ServerWebExchange request) {
```
Comment (pascalgrimaud) [inline generators/server/templates/src/main/java/package/web/rest/AccountResource.java.ejs:93]: can we have a unit or an integration test on this plz ?

Comment (vishal423) [reply generators/server/templates/src/main/java/package/web/rest/AccountResource.java.ejs:93]: sure. I will check if we have an existing test case as this API is already exposed in `Session`, `JWT` authentication scenarios.

**Review by pascalgrimaud — APPROVED** [review]

File: generators/server/templates/src/test/java/package/web/rest/AccountResourceIT_oauth2.java.ejs [hunk 2]
```diff
@@ -160,4 +160,44 @@ class AccountResourceIT <% if (databaseType === 'cassandra') { %>extends Abstrac
...
+    @WithUnauthenticatedMockUser
+    <%_ if (!reactive) { _%>
+    void testNonAuthenticatedUser() <% if (!reactive) { %>throws Exception <% } %>{
+        restAccountMockMvc.perform(get("/api/authenticate")
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(content().string(""));
+    <%_ } else { _%>
+    void testNonAuthenticatedUser() {
+        accountWebTestClient.get().uri("/api/authenticate")
```
Comment (DanielFran) [inline generators/server/templates/src/test/java/package/web/rest/AccountResourceIT_oauth2.java.ejs:174]:
```suggestion
        webTestClient.get().uri("/api/authenticate")
```

File: generators/server/templates/src/test/java/package/web/rest/AccountResourceIT_oauth2.java.ejs [hunk 3]
```diff
@@ -160,4 +160,44 @@ class AccountResourceIT <% if (databaseType === 'cassandra') { %>extends Abstrac
...
+            .with(request -> {
+                request.setRemoteUser(TEST_USER_LOGIN);
+                return request;
+            })
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(content().string(TEST_USER_LOGIN));
+    <%_ } else { _%>
+    void testAuthenticatedUser() {
+        accountWebTestClient
```
Comment (DanielFran) [inline generators/server/templates/src/test/java/package/web/rest/AccountResourceIT_oauth2.java.ejs:195]:
```suggestion
        webTestClient
```

File: generators/server/templates/src/test/java/package/web/rest/AccountResourceIT_skipUserManagement.java.ejs [hunk 4]
```diff
@@ -99,4 +99,44 @@ class AccountResourceIT <% if (databaseType === 'cassandra') { %>extends Abstrac
...
+    @WithUnauthenticatedMockUser
+    <%_ if (!reactive) { _%>
+    void testNonAuthenticatedUser() <% if (!reactive) { %>throws Exception <% } %>{
+        restAccountMockMvc.perform(get("/api/authenticate")
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(content().string(""));
+    <%_ } else { _%>
+    void testNonAuthenticatedUser() {
+        accountWebTestClient.get().uri("/api/authenticate")
```
Comment (DanielFran) [inline generators/server/templates/src/test/java/package/web/rest/AccountResourceIT_skipUserManagement.java.ejs:113]:
```suggestion
        webTestClient.get().uri("/api/authenticate")
```

File: generators/server/templates/src/test/java/package/web/rest/AccountResourceIT_skipUserManagement.java.ejs [hunk 5]
```diff
@@ -99,4 +99,44 @@ class AccountResourceIT <% if (databaseType === 'cassandra') { %>extends Abstrac
...
+            .with(request -> {
+                request.setRemoteUser(TEST_USER_LOGIN);
+                return request;
+            })
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andExpect(content().string(TEST_USER_LOGIN));
+    <%_ } else { _%>
+    void testAuthenticatedUser() {
+        accountWebTestClient
```
Comment (DanielFran) [inline generators/server/templates/src/test/java/package/web/rest/AccountResourceIT_skipUserManagement.java.ejs:134]:
```suggestion
        webTestClient
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by DanielFran** [comment]
@vishal423 Can you look at the npm tests?

**Comment by vishal423** [comment]
@DanielFran , thanks for fixing issues. I will take a look at remaining as soon as I get some time.

**Comment by vishal423** [comment]
@DanielFran , build seems to be good now.

**Review by DanielFran — APPROVED** [review]

**Pull request merged**